### PR TITLE
Add implicit bottom drag

### PR DESCRIPTION
**Pull request by hyungyukang**
This PR adds implicit bottom drag support to Omega.

- Applied implicit bottom drag as a bottom boundary condition in the implicit vertical mixing solver
- Updated bottom drag configuration options in `config/Default.yml`
```
    BottomDragTendency:
      Enable: true
      Mode: Implicit
      Type: Constant
      BottomDragCoeff: 1.0e-3
```
- Added a CTest for implicit bottom drag using the explicit bottom drag test setup
- Updated related documentation
- Renamed the existing `BottomDrag` to `ExplicitBottomDrag` for clearer separation between explicit and implicit bottom drag

Checklist
* [x] Documentation:
  * [x] [User's Guide](https://github.com/E3SM-Project/Omega/tree/develop/components/omega/doc/userGuide) has been updated
  * [x] Documentation has been [built locally](https://portal.nersc.gov/project/e3sm/hgkang/Omega/implicit_bottom_drag/html/) and changes look as expected
* Linting
  * [x] [Pre-commit](https://docs.e3sm.org/Omega/Omega/devGuide/Linting.html) run locally
* Building
  * [x] CMake build does not produce any new warnings from changes in this PR
* [ ] Testing
  **frontier, craygnu, mpich**
  - [x] CTests Pass
  - [x] Polaris omega_pr Pass (with expected diffs)

  **frontier, craygnu-mphipcc, mpich**
  - [x] CTests Pass
  - [x] Polaris omega_pr Pass (with expected diffs)

  **pm-cpu, gnu, mpich**
  - [x] CTests Pass
  - [ ] Polaris omega_pr Pass

  **pm-gpu, gnugpu, mpich**
  - [x] CTests Pass
  - [ ] Polaris omega_pr Pass

* [x] Provide relevant details in a comment to the PR titled `Testing` with the following:
  * Which machines [CTest unit tests](https://docs.e3sm.org/Omega/Omega/devGuide/QuickStart.html#running-ctests)
        have been run on and indicate that are all passing.
  * The [Polaris omega_pr test suite](https://docs.e3sm.org/Omega/Omega/devGuide/Testing.html)
     has passed, using the Polaris `e3sm_submodules/Omega` baseline
  * Document machine(s), compiler(s), and the build path(s) used for `-p` for both the baseline (Polaris `e3sm_submodules/Omega`) and the PR build
  * Indicate "All tests passed" or document failing tests
  * Document testing used to verify the changes including any tests that are added/modified/impacted.

### CODE REVIEW

**Review by mark-petersen — APPROVED** [review]
Repeated cTests on frontier with gnu on cpu and gpu, as above. Tested polaris omega_pr suite on frontier with cpu and gpu using https://github.com/E3SM-Project/polaris/pull/659. This runs `Mode: Implicit` bottom drag on 21 tests. The overflow images above are convincing. Thanks, Hyun!

File: components/omega/src/ocn/Tendencies.cpp [hunk 1]
```diff
@@ -238,13 +238,52 @@ void Tendencies::readConfig(Config *OmegaConfig ///< [in] Omega config
...
+                          "found in TendConfig");
+
+   if (BottomDragType != "Constant" && BottomDragType != "constant") {
+      ABORT_ERROR("Tendencies: BottomDragTendency Type should be one of "
+                  " 'Constant' but got {}:",
+                  BottomDragType);
+   }
+
+   this->ExplicitBottomDrag.Enabled                      = false;
+   this->VMix->VelVertMixSetup.ImplicitBottomDragEnabled = false;
```
Comment (mwarusz) [inline components/omega/src/ocn/Tendencies.cpp]:
```suggestion
```
I don't think these lines are needed. The initial values should already be `false` from looking at the constructors.

**Review by mwarusz — APPROVED** [review]
Looks good to me. I have only one minor comment. CTests passed on aurora with `oneapi-ifx`. I also tested the changes by running the `overflow` test manually. Approving.
